hypervisor/src/hypervisor/partition.rs

```diff
@@ -128,76 +140,83 @@ impl Run {
+                        fstype: Some("tmpfs".into()),
+                        flags: MsFlags::empty(),
+                        data: Some("size=500k".into()),
+                        is_dir: true,
+                    },
+                    // Mount binary
+                    FileMountInfo {
+                        source: Some(base.bin.clone()),
+                        target: "bin".into(),
+                        fstype: None,
+                        flags: MsFlags::MS_RDONLY | MsFlags::MS_BIND,
+                        data: None,
+                        is_dir: false,
+                    },
+                    // Mount /dev/null (for stdio::null)
+                    FileMountInfo {
+                        source: Some("/dev/null".into()),
+                        target: "dev/null".into(),
+                        fstype: None,
+                        flags: MsFlags::MS_RDONLY | MsFlags::MS_BIND,
+                        data: None,
+                        is_dir: false,
+                    },
+                    // Mount proc
+                    FileMountInfo {
+                        source: Some("/proc".into()),
+                        target: "proc".into(),
+                        fstype: Some("proc".into()),
+                        flags: MsFlags::empty(),
+                        data: None,
+                        is_dir: true,
+                    },
+                    // Mount CGroup v2
+                    FileMountInfo {
+                        source: None,
+                        target: "sys/fs/cgroup".into(),
+                        fstype: Some("cgroup2".into()),
+                        flags: MsFlags::empty(),
+                        data: None,
+                        is_dir: true,
+                    },
+                ];
+
+                for m in mounts {
+                    debug!("mounting {:?}", &m);
+
+                    let target: &PathBuf = &base.working_dir.path().join(m.target);
+                    let fstype = m.fstype.map(|x| PathBuf::from(x));
+                    let data = m.data.map(|x| PathBuf::from(x));
+
+                    if m.is_dir {
+                        trace!("Creating directory {}", target.display());
+                        fs::create_dir_all(target).unwrap();
+                    } else {
+                        let parent = target.parent().unwrap();
+                        trace!("Creating directory {}", parent.display());
+                        fs::create_dir_all(parent).unwrap();
+
+                        trace!("Creating file {}", target.display());
+                        fs::File::create(target).unwrap();
+                    }
+
+                    mount::<PathBuf, PathBuf, PathBuf, PathBuf>(
+                        m.source.as_ref(),
+                        target,
+                        fstype.as_ref(),
+                        m.flags,
+                        data.as_ref(),
+                    )
+                    .unwrap();
```

Comment:
Maybe it is a good Idea to attach this function directly to the `FileMountInfo` struct as an `impl` block.  
If we do this, we can also think about renaming the struct to something like `FileMount` or `FileMounter` with a self-consuming `mount` function

Reply:
This sounds like a good idea to me.
As a former C developer, I suffer from writing C in every language I touch :smile: